### PR TITLE
remove the hard coded "user" 

### PR DESCRIPTION
**Pull request by abalad**
Closes https://github.com/feathersjs-ecosystem/feathers-authentication-hooks/issues/49, closes https://github.com/feathersjs-ecosystem/feathers-authentication-hooks/issues/43

### CODE REVIEW

**Comment by abalad** [comment]
I'm having trouble trying to run the tests::

```shell
D:\Labs\Forks\FeatherAuthenticationHook\node_modules\.bin\_mocha.CMD:1
(function (exports, require, module, __filename, __dirname) { @IF EXIST "%~dp0\node.exe" (
                                                              ^

SyntaxError: Invalid or unexpected token
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:607:28)
    at Module.replacementCompile (D:\Labs\Forks\FeatherAuthenticationHook\node_modules\append-transform\index.js:58:13)
    at Module._extensions..js (module.js:654:10)
    at Object.<anonymous> (D:\Labs\Forks\FeatherAuthenticationHook\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Function.Module.runMain (module.js:684:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! feathers-authentication-hooks@0.3.0 coverage: `istanbul cover _mocha -- --opts mocha.opts`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the feathers-authentication-hooks@0.3.0 coverage script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Maicon\AppData\Roaming\npm-cache\_logs\2018-08-21T13_28_02_758Z-debug.log
npm ERR! Test failed.  See above for more details.

```

File: lib/associate-current-user.js [hunk 1]
```diff
@@ -11,17 +11,19 @@ module.exports = function (options = {}) {
       throw new Error(`The 'associateCurrentUser' hook should only be used as a 'before' hook.`);
     }
 
-    if (!hook.params.user) {
+    options = Object.assign({}, defaults, hook.app.get('authentication'), options);
+    const userEntity = hook.params[options.entity] || 'user';
```
[the remark this answers is not in the document]
Comment (abalad) [reply lib/associate-current-user.js]: OMG, Truth was my fault. Excuse me.

**Comment by abalad** [comment]
I resubmitted the run and now only 1 test is left to pass. What can it be ?

**Comment by daffl** [comment]
I fixed the text. `context.app` was `undefined` there. Thanks for adding this!

**Pull request merged**